eng/common/scripts/Helpers/Package-Helpers.ps1

```diff
@@ -68,27 +70,29 @@ Get-Content -Raw path/to/file.yml | CompatibleConvertFrom-Yaml
 #>
 function CompatibleConvertFrom-Yaml {
   param(
-    [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+    [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
     [string]$Content
   )
 
   if (!($Content)) {
     throw "Content to parse is a required input."
   }
 
-  # Initialize any variables or checks that need to be done once
-  $yqPresent = Get-Command 'yq' -ErrorAction SilentlyContinue
-  if (-not $yqPresent) {
-    . (Join-Path $PSScriptRoot PSModule-Helpers.ps1)
-    Install-ModuleIfNotInstalled -WhatIf:$false "powershell-yaml" "0.4.1" | Import-Module
+  if ($null -eq $script:yqPresent) {
+    $script:yqPresent = [bool](Get-Command 'yq' -ErrorAction SilentlyContinue)
+
+    if (-not $script:yqPresent) {
+      Write-Host "yqPresent evaluated to null, then false."
+      . (Join-Path $PSScriptRoot PSModule-Helpers.ps1)
+      Install-ModuleIfNotInstalled -WhatIf:$false "powershell-yaml" "0.4.1" | Import-Module
```

Comment:
This won't fix the issue, as if yq isn't present we'll still do the yaml install in a loop. I think we need a separate variable like `WE_INSTALLED_THE_PS_MODULE` so we can skip duplicate calls altogether, even though they are idempotent they take time.

Reply:
I've run this it absolutely fixes the issue.

Reply:
I'm going to keep this, as it eliminates the install loop. I'll update this PR to remove the `yq` Usage after #9392 merges